### ISSUE TITLE
[Feature Request] Allow custom merge commit messages

### ISSUE THREAD

**Issue by lkster** [ForkIssues/TrackerWin#2018]
In case of eg. squash merge with conventional commits approach commit messages should have specific structure such as `feat(scope): some desc`. It's currently not possible because of commit message being not possible to be changed. 

Current workaround is merging and then amending commit message.

**Comment by DanPristupov** [comment]
You can merge with `--no-commit` option and Fork will bring you to the message editing.

https://github.com/fork-dev/TrackerWin/issues/244

**Comment by lkster** [comment]
Yes, but this is not applicable in case of squash merge. I.e. there will be two parents while squash merge is about dumping working branch to have linear history. Unless I just didn't catch the option to perform `--no-commit --squash`

**Comment by DanPristupov** [comment]
No, selecting two options in the dropdown is not possible. So, using amend is the only option to edit message for squash merge.

**Comment by lkster** [comment]
So is there any chance to have custom message textbox or anything reconsidered?

**Comment by DanPristupov** [comment]
Oh, no, most likely not. This will make merge look too complicated to anyone who doesn't need custom messaging. I want to keep the UI simple.

**Comment by ddbtrmatic** [comment]
FWIW, the maintainers of conventional commits recommend not using merge commits at all.

https://github.com/conventional-commits/conventionalcommits.org/issues/454#issuecomment-1136024464

**Comment by lkster** [comment]
> Oh, no, most likely not. This will make merge look too complicated to anyone who doesn't need custom messaging. I want to keep the UI simple.

Actually simple "Custom commit message" checkbox which after checking would show up textarea below (prefilled with default mesage) shouldn't make look & feel more complicated in any way. So everyone not insterested in changing commit message would just have that checkbox unchecked (which would be it's default state). Eg. Azure DevOps has it this way

**Comment by henkhenken** [comment]
> Actually simple "Custom commit message" checkbox which after checking would show up textarea below (prefilled with default mesage) shouldn't make look & feel more complicated in any way. So everyone not insterested in changing commit message would just have that checkbox unchecked (which would be it's default state). Eg. Azure DevOps has it this way

This sounds like a simple and a great addition to the application. I hope it can be added soon.
